SpatialGDK/Source/SpatialGDKFunctionalTests/Public/AutoDestroyComponent.h

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Components/ActorComponent.h"
+#include "AutoDestroyComponent.generated.h"
+
+
+/*
+* Empty component to be added to actors so that they can be automatically destroyed when the tests finish
+*/
+UCLASS( ClassGroup=(Custom), meta=(BlueprintSpawnableComponent) )
+class SPATIALGDKFUNCTIONALTESTS_API UAutoDestroyComponent : public USceneComponent
```

Comment:
It is in the SPATIALGDKFUNCTIONALTESTS_API is it also necessary to add SpatialFunctionalTest in the component name as it makes it very long?

Reply:
The API part will handle how it gets bundled into dlls, so you need that. Then when somebody is using the engine, if you have a component that is called AutoDestroyComponent it seems like something you can use any time. I know the names are long as hell, but at least descriptive that they are meant to be used in that context only

Reply:
How about adding this to the class definition UCLASS( NotBlueprintable, ClassGroup=SpatialFunctionalTest ). Then this class cannot be used in the Blueprint and the group makes it clear that it is part of the spatial functional test framework?

Reply:
I think atm it would be better to follow the way I mentioned to keep consistency with the whole framework. However, we can then do a PR to change the whole system to be like that, I'd have to check the ClassGroup. Also keep in mind that some classes need to be exposed to BP, like ASpatialFunctionalTestFlowController.

Reply:
Ok

Reply:
Done